services/contentful.js

```diff
@@ -140,23 +140,31 @@ module.exports.getCoursesByCategory = assert((category, locale = 'en-US', api =
     include: 6
   })
     .then((response) => response.items)
-}, 'Category')
+})
 
 // Utility function
 function getClient (api = 'cda') {
   return api === 'cda' ? deliveryClient : previewClient
 }
 
-function assert (fn, context) {
-  return function (request, response, next) {
-    return fn(request, response, next)
-    .then((data) => {
-      if (!data) {
-        var err = new Error(`${context} Not Found`)
-        err.status = 404
-        throw err
-      }
-      return data
-    })
+/**
+ * Utility function for wrapping regular API calls.
+ * This is done for easily catching 404 errors.
+ * @param  {string}   context The type of result the function is looking for
+ * @param  {Function} fn      The function to wrap
+ * @return {Object}           The result of `fn`, if not empty
+ * @throws {Error}    When `fn` returns an empty result
+ */
+function throwOnEmptyResult (context, fn) {
+  return function (...params) {
+    return fn(...params)
+      .then((data) => {
```

Comment:
await here and turn the function in to an async one plz

Reply:
I'd rather not tbh, I just changed the name of the function to make it more readable and replaced the parameter names (which were incorrect) with a simple `..params`. Changing the actual behavior of the function is beyond the scope of this PR.

Reply:
U are right here, just realised we have a mix of styles in here. Mostly async/await but there are a few pure promises around. Out of scope ✅